mlflow/tracking/_tracking_service/client.py

```diff
@@ -216,29 +220,65 @@ def get_trace_info(self, request_id):
         """
         return self.store.get_trace_info(request_id)
 
-    def search_traces(
+    def _search_traces(
         self,
         experiment_ids: List[str],
         filter_string: Optional[str] = None,
         max_results: int = SEARCH_TRACES_DEFAULT_MAX_RESULTS,
         order_by: Optional[List[str]] = None,
         page_token: Optional[str] = None,
-    ) -> PagedList[Trace]:
-        trace_infos, token = self.store.search_traces(
+    ):
+        return self.store.search_traces(
             experiment_ids=experiment_ids,
             filter_string=filter_string,
             max_results=max_results,
             order_by=order_by,
             page_token=page_token,
         )
-        with ThreadPoolExecutor() as executor:
-            traces = executor.map(
-                lambda ti: Trace(
-                    trace_info=ti,
-                    trace_data=self._download_trace_data(ti.request_id),
-                ),
-                trace_infos,
+
+    def search_traces(
+        self,
+        experiment_ids: List[str],
+        filter_string: Optional[str] = None,
+        max_results: int = SEARCH_TRACES_DEFAULT_MAX_RESULTS,
+        order_by: Optional[List[str]] = None,
+        page_token: Optional[str] = None,
+    ) -> PagedList[Trace]:
+        def fn(trace_info: TraceInfo) -> Optional[TraceData]:
+            try:
+                trace_data = self._download_trace_data(trace_info.request_id)
+            except Exception:
+                _logger.debug(
+                    "Failed to download trace data for trace with request_id=%s",
+                    trace_info.request_id,
+                    exc_info=True,
+                )
```

Comment:
Why not log a warning message too ? By default debug level log is disabled so user does not see it.

Reply:
What if you have 10 failed downloads?

Reply:
Can we make it to show only one warning if failure count > 1 ?

Reply:
Assuming all downloads failed with the same reason?

Reply:
I think we can emit one warning like:

```
Failed to download trace data for trace with request_id={id list}, you can enable DEBUG level logging for getting details.
```

Reply:
I'm still leaning towards `debug`.

@dbczumar WDYT?

Reply:
I agree with `debug` here. There isn't anything that the user can do to fix this, so why warn?